src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -219,8 +220,14 @@ internal static ImmutableArray<ParameterSymbol> DeriveParameters(MethodSymbol so
             foreach (var oldParam in sourceMethod.Parameters)
             {
                 //same properties as the old one, just change the owner
-                builder.Add(SynthesizedParameterSymbol.Create(destinationMethod, oldParam.TypeWithAnnotations, oldParam.Ordinal,
-                    oldParam.RefKind, oldParam.Name, oldParam.RefCustomModifiers));
+                builder.Add(Create(
+                    destinationMethod,
+                    oldParam.TypeWithAnnotations,
+                    oldParam.Ordinal,
+                    oldParam.RefKind,
+                    oldParam.Name,
+                    oldParam.RefCustomModifiers,
+                    (oldParam as SynthesizedComplexParameterSymbol)?.BaseParameterForAttributes));
```

Comment:
>(oldParam as SynthesizedComplexParameterSymbol) [](start = 20, length = 47)

Are there scenarios when this is not null?

Reply:
I don't think any test is affected by this change. This is used for SynthesizedEntryPointSymbol, SynthesizedImplementationMethod, and SynthesizedSealedPropertyAccessor. Actually, this seems like a behavior change where for example a synthesized sealed accessor may start to contain attributes from a base accessor from source, when it didn't before. So perhaps the right thing to do here is to just revert the change.

---
In reply to: [373194840](https://github.com/dotnet/roslyn/pull/41299#discussion_r373194840) [](ancestors = 373194840)

Reply:
If for all scenarios that can happen today ```oldParam``` is never a ```SynthesizedComplexParameterSymbol```, I would add an assert to confirm this fact and pass null.

---
In reply to: [373204933](https://github.com/dotnet/roslyn/pull/41299#discussion_r373204933) [](ancestors = 373204933,373194840)